crates/rolldown/src/ecmascript/ecma_generator.rs

```diff
@@ -106,18 +104,22 @@ impl Generator for EcmaGenerator {
       map.set_sources(sources.iter().map(std::convert::AsRef::as_ref).collect::<Vec<_>>());
     }
 
-    Ok(vec![PreliminaryAsset {
-      origin_chunk: ctx.chunk_idx,
-      content,
-      map,
-      meta: AssetMeta::from(EcmaAssetMeta { rendered_chunk }),
-      augment_chunk_hash: None,
-      file_dir: file_dir.to_path_buf(),
-      preliminary_filename: ctx
-        .chunk
-        .preliminary_filename
-        .clone()
-        .expect("should have preliminary filename"),
-    }])
+    Ok(GenerateOutput {
+      assets: vec![PreliminaryAsset {
+        origin_chunk: ctx.chunk_idx,
+        content,
+        map,
+        meta: AssetMeta::from(EcmaAssetMeta { rendered_chunk }),
+        augment_chunk_hash: None,
+        file_dir: file_dir.to_path_buf(),
+        preliminary_filename: ctx
+          .chunk
+          .preliminary_filename
+          .clone()
+          .expect("should have preliminary filename"),
+      }],
+      errors: std::mem::take(&mut ctx.errors),
+      warnings: std::mem::take(&mut ctx.warnings),
```

Comment:
Since we already stored the warnings in the context. Why do we need to take it out and return it in other places?

Reply:
Here is not same warnings. The render_chunk_to_assets warnings come from link output. Here render_preliminary_assets is run in parallel, we can't using `LinkStageOutput#warnings`.